### PR TITLE
test: add missing vaadin-testbench-loadtest unit tests

### PR DESCRIPTION
**Pull request by tltv**
Adds unit tests for public API in testbench-loadtest-support and testbench-converter-plugin.

PartOf: #2210

### CODE REVIEW

File: vaadin-testbench-loadtest/testbench-converter-plugin/src/test/java/com/vaadin/testbench/loadtest/util/K6ScenarioCombinerTest.java [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+
+        K6ScenarioCombiner combiner = new K6ScenarioCombiner();
+        Path output = tempDir.resolve("combined.js");
+
+        combiner.combine(List.of(new ScenarioConfig("alpha", a, 50),
+                new ScenarioConfig("beta", b, 50)), output, 10, "30s");
+
+        String content = Files.readString(output);
+
+        // Each scenario gets floor(50 * 10 / 100) = 5 VUs.
```
Comment (caalador) [inline vaadin-testbench-loadtest/testbench-converter-plugin/src/test/java/com/vaadin/testbench/loadtest/util/K6ScenarioCombinerTest.java]: This only checks that the named scenarios are there, but it doesn't check any stage that would show that they both get a target of 5 VUs
Would expect some kind of check on the stages:
```
      stages: [
        { duration: '10s', target: 5 },
        { duration: '10s', target: 5 },
        { duration: '10s', target: 0 },
      ],
```
Also perhaps verify that the default executor is ramping-vus

File: vaadin-testbench-loadtest/testbench-converter-plugin/src/test/java/com/vaadin/testbench/loadtest/util/K6ScenarioCombinerTest.java [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+
+        String content = Files.readString(output);
+        // small gets ~25 VUs (1/4 * 100), big gets ~75 (3/4 * 100).
+        assertTrue(content.contains("small: {"));
+        assertTrue(content.contains("big: {"));
+        // The scenario named "big" must not collide with "small".
+        int smallExec = content.indexOf("exec: 'smallScenario'");
+        int bigExec = content.indexOf("exec: 'bigScenario'");
+        assertTrue(smallExec >= 0);
+        assertTrue(bigExec >= 0);
```
Comment (caalador) [inline vaadin-testbench-loadtest/testbench-converter-plugin/src/test/java/com/vaadin/testbench/loadtest/util/K6ScenarioCombinerTest.java]: This also doesn't really check that the scenarios do not collide. Only checks that they are found in the content.

File: vaadin-testbench-loadtest/testbench-converter-plugin/src/test/java/com/vaadin/testbench/loadtest/util/K6ScenarioCombinerTest.java [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+        K6ScenarioCombiner combiner = new K6ScenarioCombiner();
+        Path output = tempDir.resolve("combined.js");
+
+        combiner.combine(List.of(new ScenarioConfig("small", small, 1),
+                new ScenarioConfig("big", big, 3)), output, 100, "30s");
+
+        String content = Files.readString(output);
+        // small gets ~25 VUs (1/4 * 100), big gets ~75 (3/4 * 100).
+        assertTrue(content.contains("small: {"));
+        assertTrue(content.contains("big: {"));
```
Comment (caalador) [inline vaadin-testbench-loadtest/testbench-converter-plugin/src/test/java/com/vaadin/testbench/loadtest/util/K6ScenarioCombinerTest.java]: Doesn't check the VUs in any way here. Only that the scenarios small and big can be found.

File: vaadin-testbench-loadtest/testbench-converter-plugin/src/test/java/com/vaadin/testbench/loadtest/util/K6TestRefactorerTest.java [hunk 4]
```diff
@@ -8,13 +8,144 @@
...
+            """;
+
+    @Test
+    void refactorContent_addsVaadinHelperImport() {
+        K6TestRefactorer refactorer = new K6TestRefactorer(
+                ThinkTimeConfig.DISABLED);
+
+        String refactored = refactorer.refactorContent(MINIMAL_SCRIPT);
+
+        assertTrue(refactored.contains("vaadin-k6-helpers.js"),
```
Comment (caalador) [inline vaadin-testbench-loadtest/testbench-converter-plugin/src/test/java/com/vaadin/testbench/loadtest/util/K6TestRefactorerTest.java]: could be at least `from '../utils/vaadin-k6-helpers.js'` to know it's used in an import format instead of for instance a comment

File: vaadin-testbench-loadtest/testbench-converter-plugin/src/test/java/com/vaadin/testbench/loadtest/util/K6TestRefactorerTest.java [hunk 5]
```diff
@@ -8,13 +8,144 @@
...
+    @Test
+    void refactorContent_addsBaseUrlConstants() {
+        K6TestRefactorer refactorer = new K6TestRefactorer(
+                ThinkTimeConfig.DISABLED);
+
+        String refactored = refactorer.refactorContent(MINIMAL_SCRIPT);
+
+        assertTrue(refactored.contains("const APP_IP"));
+        assertTrue(refactored.contains("const APP_PORT"));
+        assertTrue(refactored.contains("const BASE_URL"));
```
Comment (caalador) [inline vaadin-testbench-loadtest/testbench-converter-plugin/src/test/java/com/vaadin/testbench/loadtest/util/K6TestRefactorerTest.java]: As they shouldn't change could check for full constant  instead
```
const APP_IP = __ENV.APP_IP || 'localhost';
const APP_PORT = __ENV.APP_PORT || '8080';
const BASE_URL = `http://${APP_IP}:${APP_PORT}`;
```

File: vaadin-testbench-loadtest/testbench-converter-plugin/src/test/java/com/vaadin/testbench/loadtest/util/K6TestRefactorerTest.java [hunk 6]
```diff
@@ -8,13 +8,144 @@
...
+                  // Another uidl request
+                  http.post('http://localhost:8080/?v-r=uidl', '{}')
+                }
+                """;
+
+        String refactored = refactorer.refactorContent(scriptWithUidl);
+
+        assertTrue(refactored.contains("// Think time:"),
+                "Expected think time comment when enabled with uidl blocks:\n"
+                        + refactored);
```
Comment (caalador) [inline vaadin-testbench-loadtest/testbench-converter-plugin/src/test/java/com/vaadin/testbench/loadtest/util/K6TestRefactorerTest.java:138]: This only checks that we have a comment starting as Think time: and not that the actual sleep is there.

File: vaadin-testbench-loadtest/testbench-converter-plugin/src/test/java/com/vaadin/testbench/loadtest/util/ResourceExtractorTest.java [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+
+        assertEquals(extractDir, result);
+        assertTrue(Files.exists(extractor.getVaadinHelpersScript()));
+        assertTrue(Files.exists(extractor.getK6SummaryScript()));
+        assertTrue(Files.size(extractor.getVaadinHelpersScript()) > 0);
+        assertTrue(Files.size(extractor.getK6SummaryScript()) > 0);
+    }
+
+    @Test
+    void extractUtilities_isIdempotent() throws IOException {
```
Comment (caalador) [inline vaadin-testbench-loadtest/testbench-converter-plugin/src/test/java/com/vaadin/testbench/loadtest/util/ResourceExtractorTest.java]: Not certain this is needed as the resource is loaded from the classLoader and that should work the same each time.

File: vaadin-testbench-loadtest/testbench-converter-plugin/src/test/java/com/vaadin/testbench/loadtest/util/ResourceExtractorTest.java [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+        extractor.extractUtilities();
+        long firstSize = Files.size(extractor.getVaadinHelpersScript());
+        extractor.extractUtilities();
+        long secondSize = Files.size(extractor.getVaadinHelpersScript());
+
+        assertEquals(firstSize, secondSize);
+    }
+
+    @Test
+    void getExtractionDir_returnsConstructorValue() {
```
Comment (caalador) [inline vaadin-testbench-loadtest/testbench-converter-plugin/src/test/java/com/vaadin/testbench/loadtest/util/ResourceExtractorTest.java]: This test also basically tests java basic functions `this.extractionDir = extractionDir;`  and `return extractionDir;`

File: vaadin-testbench-loadtest/testbench-converter-plugin/src/test/java/com/vaadin/testbench/loadtest/util/ResourceExtractorTest.java [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+        extractor.extractUtilities();
+        assertTrue(Files.exists(extractDir));
+
+        extractor.cleanup();
+
+        assertFalse(Files.exists(extractDir));
+    }
+
+    @Test
+    void cleanup_onMissingDir_doesNotThrow() throws IOException {
```
Comment (caalador) [inline vaadin-testbench-loadtest/testbench-converter-plugin/src/test/java/com/vaadin/testbench/loadtest/util/ResourceExtractorTest.java:54]: This also feels quite unnecessary as it tests Files.exists, but guess the method could use a faulty delete function

File: vaadin-testbench-loadtest/testbench-converter-plugin/src/test/java/com/vaadin/testbench/loadtest/util/SourceHasherTest.java [hunk 10]
```diff
@@ -0,0 +1,121 @@
...
+
+        assertEquals("abc123", hasher.readStoredHash(file));
+    }
+
+    @Test
+    void findsTestClassInItSourceLocation() throws IOException {
+        Path dir = workDir.resolve("src/it/java/com/example");
+        Files.createDirectories(dir);
+        Files.writeString(dir.resolve("ItOnly.java"), "class ItOnly {}");
+        writePom("<project/>");
```
Comment (caalador) [inline vaadin-testbench-loadtest/testbench-converter-plugin/src/test/java/com/vaadin/testbench/loadtest/util/SourceHasherTest.java]: What is the pom needed for? It is added to the digest, but it is not needed for finding test files or generating the hash.

File: vaadin-testbench-loadtest/testbench-loadtest-support/src/test/java/com/vaadin/testbench/loadtest/K6RecordingExtensionTest.java [hunk 11]
```diff
@@ -0,0 +1,180 @@
...
+
+        K6RecordingExtension extension = new K6RecordingExtension();
+        ConditionEvaluationResult result = extension
+                .evaluateExecutionCondition(context);
+
+        assertFalse(result.isDisabled());
+    }
+
+    @Test
+    void afterEach_instanceNotHasDriver_noop() throws Exception {
```
Comment (caalador) [inline vaadin-testbench-loadtest/testbench-loadtest-support/src/test/java/com/vaadin/testbench/loadtest/K6RecordingExtensionTest.java]: Strange naming hard to understand. `noDriver_noExceptionForAfterEach` would be more descriptive than NotHasDriver_noop

File: vaadin-testbench-loadtest/testbench-loadtest-support/src/test/java/com/vaadin/testbench/loadtest/K6RecordingExtensionTest.java [hunk 12]
```diff
@@ -0,0 +1,180 @@
...
+        ExtensionContext context = Mockito.mock(ExtensionContext.class);
+        Mockito.when(context.getRequiredTestInstance())
+                .thenReturn("notADriver");
+
+        K6RecordingExtension extension = new K6RecordingExtension();
+        assertDoesNotThrow(() -> extension.afterEach(context));
+    }
+
+    @Test
+    void afterEach_hasDriverWithNullDriver_noop() throws Exception {
```
Comment (caalador) [inline vaadin-testbench-loadtest/testbench-loadtest-support/src/test/java/com/vaadin/testbench/loadtest/K6RecordingExtensionTest.java]: `nullDriver_noExceptionForAfterEach` would show the test state and expectation better.

File: vaadin-testbench-loadtest/testbench-loadtest-support/src/test/java/com/vaadin/testbench/loadtest/K6RecordingExtensionTest.java [hunk 13]
```diff
@@ -0,0 +1,180 @@
...
+        HasDriverFake fake = new HasDriverFake(null);
+        ExtensionContext context = Mockito.mock(ExtensionContext.class);
+        Mockito.when(context.getRequiredTestInstance()).thenReturn(fake);
+
+        K6RecordingExtension extension = new K6RecordingExtension();
+        assertDoesNotThrow(() -> extension.afterEach(context));
+    }
+
+    @Test
+    void afterEach_hasDriver_quitsDriver() throws Exception {
```
Comment (caalador) [inline vaadin-testbench-loadtest/testbench-loadtest-support/src/test/java/com/vaadin/testbench/loadtest/K6RecordingExtensionTest.java]: `driverQuitCalled_whenAfterEachCompletes`

File: vaadin-testbench-loadtest/testbench-loadtest-support/src/test/java/com/vaadin/testbench/loadtest/K6RecordingExtensionTest.java [hunk 14]
```diff
@@ -0,0 +1,180 @@
...
+        Mockito.when(context.getRequiredTestInstance()).thenReturn(fake);
+
+        K6RecordingExtension extension = new K6RecordingExtension();
+        extension.afterEach(context);
+
+        Mockito.verify(driver).quit();
+    }
+
+    @Test
+    void afterEach_quitThrows_swallowed() throws Exception {
```
Comment (caalador) [inline vaadin-testbench-loadtest/testbench-loadtest-support/src/test/java/com/vaadin/testbench/loadtest/K6RecordingExtensionTest.java]: `driverQuitThrows_noExceptionFromAfterEach`

File: vaadin-testbench-loadtest/testbench-loadtest-support/src/test/java/com/vaadin/testbench/loadtest/LoadTestMetricsTest.java [hunk 15]
```diff
@@ -0,0 +1,51 @@
...
+class LoadTestMetricsTest {
+
+    @Test
+    void newInstance_activeUiCountIsZero() {
+        LoadTestMetrics metrics = new LoadTestMetrics();
+        assertEquals(0, metrics.getActiveUiCount());
+    }
+
+    @Test
+    void newInstance_doesNotThrow() {
```
Comment (caalador) [inline vaadin-testbench-loadtest/testbench-loadtest-support/src/test/java/com/vaadin/testbench/loadtest/LoadTestMetricsTest.java]: This test seems strange. Why would there be an expectation that the constructor might throw, also how does it cover both branches?

I would remove this test.

File: vaadin-testbench-loadtest/testbench-loadtest-support/src/test/java/com/vaadin/testbench/loadtest/LoadTestMetricsTest.java [hunk 16]
```diff
@@ -0,0 +1,51 @@
...
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+class LoadTestMetricsTest {
+
+    @Test
+    void newInstance_activeUiCountIsZero() {
+        LoadTestMetrics metrics = new LoadTestMetrics();
+        assertEquals(0, metrics.getActiveUiCount());
```
Comment (caalador) [inline vaadin-testbench-loadtest/testbench-loadtest-support/src/test/java/com/vaadin/testbench/loadtest/LoadTestMetricsTest.java]: This test also tests that empty set is empty. Perhaps the ActiveUiCount should just be checked in the trackUI test before adding UI, except that also doesn't increment the UI if afterNavigationEvent is not fired for the UI which would need some more mocking on the mockUI

File: vaadin-testbench-loadtest/testbench-loadtest-support/src/test/java/com/vaadin/testbench/loadtest/LoadTestServiceInitListenerTest.java [hunk 17]
```diff
@@ -0,0 +1,31 @@
...
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import com.vaadin.flow.server.ServiceInitEvent;
+import com.vaadin.flow.server.VaadinService;
+
+class LoadTestServiceInitListenerTest {
+
+    @Test
+    void serviceInit_registersSessionAndUiInitListeners() {
```
Comment (caalador) [inline vaadin-testbench-loadtest/testbench-loadtest-support/src/test/java/com/vaadin/testbench/loadtest/LoadTestServiceInitListenerTest.java:20]: Not certain about how useful this test is. Basically it only verifies that addInitListeners are called

**Review by caalador — CHANGES_REQUESTED** [review]

File: vaadin-testbench-loadtest/testbench-loadtest-support/src/test/java/com/vaadin/testbench/loadtest/LoadTestServiceInitListenerTest.java (hunk 17 above)
Comment (tltv) [reply vaadin-testbench-loadtest/testbench-loadtest-support/src/test/java/com/vaadin/testbench/loadtest/LoadTestServiceInitListenerTest.java:20]: The listener genuinely has no behavior beyond registering those two listeners.

File: vaadin-testbench-loadtest/testbench-converter-plugin/src/test/java/com/vaadin/testbench/loadtest/util/ResourceExtractorTest.java (hunk 8 above)
Comment (tltv) [reply vaadin-testbench-loadtest/testbench-converter-plugin/src/test/java/com/vaadin/testbench/loadtest/util/ResourceExtractorTest.java]: Kept it since it still tests the public API to not throw on cleanup with missing dir.

Edit: this was removed. Commented in wrong line.

File: vaadin-testbench-loadtest/testbench-converter-plugin/src/test/java/com/vaadin/testbench/loadtest/util/K6ScenarioCombinerTest.java (hunk 1 above)
Comment (tltv) [reply vaadin-testbench-loadtest/testbench-converter-plugin/src/test/java/com/vaadin/testbench/loadtest/util/K6ScenarioCombinerTest.java]: done

File: vaadin-testbench-loadtest/testbench-converter-plugin/src/test/java/com/vaadin/testbench/loadtest/util/K6ScenarioCombinerTest.java (hunk 2 above)
Comment (tltv) [reply vaadin-testbench-loadtest/testbench-converter-plugin/src/test/java/com/vaadin/testbench/loadtest/util/K6ScenarioCombinerTest.java]: Now asserts small block has target: 25 and no target: 75, big block has target: 75 and no target: 25

File: vaadin-testbench-loadtest/testbench-converter-plugin/src/test/java/com/vaadin/testbench/loadtest/util/K6TestRefactorerTest.java (hunk 4 above)
Comment (tltv) [reply vaadin-testbench-loadtest/testbench-converter-plugin/src/test/java/com/vaadin/testbench/loadtest/util/K6TestRefactorerTest.java]: done

File: vaadin-testbench-loadtest/testbench-converter-plugin/src/test/java/com/vaadin/testbench/loadtest/util/K6TestRefactorerTest.java (hunk 5 above)
Comment (tltv) [reply vaadin-testbench-loadtest/testbench-converter-plugin/src/test/java/com/vaadin/testbench/loadtest/util/K6TestRefactorerTest.java]: done

File: vaadin-testbench-loadtest/testbench-converter-plugin/src/test/java/com/vaadin/testbench/loadtest/util/K6TestRefactorerTest.java (hunk 6 above)
Comment (tltv) [reply vaadin-testbench-loadtest/testbench-converter-plugin/src/test/java/com/vaadin/testbench/loadtest/util/K6TestRefactorerTest.java:138]: done

File: vaadin-testbench-loadtest/testbench-converter-plugin/src/test/java/com/vaadin/testbench/loadtest/util/ResourceExtractorTest.java (hunk 7 above)
Comment (tltv) [reply vaadin-testbench-loadtest/testbench-converter-plugin/src/test/java/com/vaadin/testbench/loadtest/util/ResourceExtractorTest.java]: removed

File: vaadin-testbench-loadtest/testbench-converter-plugin/src/test/java/com/vaadin/testbench/loadtest/util/ResourceExtractorTest.java (hunk 9 above)
Comment (tltv) [reply vaadin-testbench-loadtest/testbench-converter-plugin/src/test/java/com/vaadin/testbench/loadtest/util/ResourceExtractorTest.java:54]: Kept it since it still tests the public API to not throw on cleanup with missing dir.

File: vaadin-testbench-loadtest/testbench-converter-plugin/src/test/java/com/vaadin/testbench/loadtest/util/SourceHasherTest.java (hunk 10 above)
Comment (tltv) [reply vaadin-testbench-loadtest/testbench-converter-plugin/src/test/java/com/vaadin/testbench/loadtest/util/SourceHasherTest.java]: removed

File: vaadin-testbench-loadtest/testbench-loadtest-support/src/test/java/com/vaadin/testbench/loadtest/LoadTestMetricsTest.java (hunk 16 above)
Comment (tltv) [reply vaadin-testbench-loadtest/testbench-loadtest-support/src/test/java/com/vaadin/testbench/loadtest/LoadTestMetricsTest.java]: Removed; moved the assertEquals(0, metrics.getActiveUiCount()) into trackUI_registersAfterNavigationAndDetachListeners.

File: vaadin-testbench-loadtest/testbench-loadtest-support/src/test/java/com/vaadin/testbench/loadtest/LoadTestMetricsTest.java (hunk 15 above)
Comment (tltv) [reply vaadin-testbench-loadtest/testbench-loadtest-support/src/test/java/com/vaadin/testbench/loadtest/LoadTestMetricsTest.java]: removed

**Review by caalador — APPROVED** [review]
